frontend/src/app/restaurant/page.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react'
+
+interface Restaurant {
+  name: string
+  address: string
+  cuisine: string
+  rating: string
+  phone: string
+  image_url: string
+  website_url: string
+}
+
+interface RestaurantProps {
+  restaurant: Restaurant
+}
+
+interface GetRestaurantsResponse {
+  data: Restaurant[]
+}
+
+const RestaurantCard: React.FC<RestaurantProps> = ({ restaurant }) => {
+  return (
+    <>
+      <h1>{restaurant.name}</h1>
+      <div>{restaurant.address}</div>
+      <div>{restaurant.cuisine}</div>
+      <div>{restaurant.rating}</div>
+      <div>{restaurant.phone}</div>
+      <div>{restaurant.image_url}</div>
+      <div>{restaurant.website_url}</div>
+      <br />
+    </>
+  )
+}
+
+const getRestaurants = async (): Promise<Restaurant[]> => {
+  const response = await fetch('http://localhost:4000/api/restaurants/all')
+
+  if (!response.ok) {
+    throw new Error('Failed to fetch restaurant data')
+  }
+
+  const restaurantData: GetRestaurantsResponse = await response.json()
+  return restaurantData.data
+}
+
+const RestaurantPage: React.FC<RestaurantProps> = async () => {
```

Comment:
Per [Next's testing docs](nextjs.org/docs/pages/building-your-application/optimizing/testing), "Follow Jest's convention by adding tests to the __tests__ folder in your project's root directory."

I agree that setting up msw should be a separate ticket, but there should be at least basic tests, for example, that `RestaurantCard` is rendering the expected information from `restaurant`, and that `RestaurantPage` is rendering the expected number of `RestaurantCards`.

Plop can scaffold your tests for you, and handle setting up the components in separate files.

Reply:
@mackenziequinnjetton the link you provided doesn't work and I couldn't find that quote. 

In the code example in this repo, if you look at the [app/client folder](https://github.com/vercel/next.js/tree/canary/examples/with-jest/app/client), there is a test in that folder as well as the `__tests__` in the root.  I also interpretted that root tests folder as tests for the root of the app only, rather than putting all tests into that root folder. 

As for tests, I'm not sure if you were part of the discussion but we talked not being strict on tests for PRs and not necessarily needing coverage for everything. It depends on what people want to focus on. 

I'm gonna defer the test since we have new issues created to display the first card. I think Karen wanted to work more on testing so I'll ask if she can add a test for seeing one restaurant.  And/or I'll create a new ticket for just the testing portion.

Reply:
I just want to push the code changes to main so others have something to work off of 😅